pupil_src/shared_modules/calibration_routines/data_processing.py

```diff
@@ -0,0 +1,203 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2020 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def filter_confidence(pupil_list, threshold):
+    len_pre_filter = len(pupil_list)
+
+    pupil_list = [p for p in pupil_list if p["confidence"] >= threshold]
+
+    len_post_filter = len(pupil_list)
+    try:
+        dismissed_percentage = 100 * (1.0 - len_post_filter / len_pre_filter)
+    except ZeroDivisionError:
+        pass  # empty pupil_list, is being handled in match_data
+    else:
+        logger.info(
+            "Dismissing {:.2f}% pupil data due to confidence < {:.2f}".format(
+                dismissed_percentage, threshold
+            )
+        )
+
+    return pupil_list
+
+
+def match_data(g_pool, pupil_list, ref_list):
+    """
+    Returns binocular and monocular matched pupil datums and ref points.
+    Uses a dispersion criterion to dismiss matches which are too far apart.
+    """
+
+    if not (pupil_list or ref_list):
+        not_enough_data_error_msg = (
+            "Not enough ref point or pupil data available for calibration."
+        )
+        logger.error(not_enough_data_error_msg)
+        return {
+            "subject": "calibration.failed",
+            "reason": not_enough_data_error_msg,
+            "timestamp": g_pool.get_timestamp(),
+            "record": True,
+        }
+
+    pupil0 = [p for p in pupil_list if p["id"] == 0]
+    pupil1 = [p for p in pupil_list if p["id"] == 1]
+
+    matched_binocular_data = closest_matches_binocular(ref_list, pupil_list)
+    matched_pupil0_data = closest_matches_monocular(ref_list, pupil0)
+    matched_pupil1_data = closest_matches_monocular(ref_list, pupil1)
+
+    if len(matched_pupil0_data) > len(matched_pupil1_data):
+        matched_monocular_data = matched_pupil0_data
+    else:
+        matched_monocular_data = matched_pupil1_data
+
+    logger.info(
+        "Collected {} monocular calibration data.".format(len(matched_monocular_data))
```

Comment:
f-strings are awesome 😉 
No need to change all `.format()` things now, but keep in mind for the future, I thinks it's more readable like this.

```suggestion
        f"Collected {len(matched_monocular_data)} monocular calibration data."
```